idl/src/build.rs

```diff
@@ -59,25 +66,30 @@ pub fn build_idl(
 }
 
 /// Build IDL.
-fn build(program_path: &Path, resolution: bool, skip_lint: bool, no_docs: bool) -> Result<Idl> {
+fn build(
+    program_path: &Path,
+    resolution: bool,
+    skip_lint: bool,
+    no_docs: bool,
+    cargo_args: &[String],
+) -> Result<Idl> {
     // `nightly` toolchain is currently required for building the IDL.
     let toolchain = std::env::var("RUSTUP_TOOLCHAIN")
         .map(|toolchain| format!("+{}", toolchain))
         .unwrap_or_else(|_| "+nightly".to_string());
 
     install_toolchain_if_needed(&toolchain)?;
-
+    let mut args = vec![
+        &toolchain,
+        "test",
+        "__anchor_private_print_idl",
+        "--features",
+        "idl-build",
+    ];
+    args.extend(cargo_args.iter().map(String::as_str));
+    args.extend(["--", "--show-output", "--quiet"]);
```

Comment:
Nit: [`args()`](https://doc.rust-lang.org/std/process/struct.Command.html#method.args) method extends the previous arguments on each call, so no need to declare separately and manually extend.

```rs
Command::new("cargo")
        .args([
            &toolchain,
            "test",
            "__anchor_private_print_idl",
            "--features",
            "idl-build",
        ])
        .args(cargo_args)
        .args(["--", "--show-output", "--quiet"])
```